utilities/test_suite/HIP/Tensor_hip.cpp

```diff
@@ -0,0 +1,1877 @@
+    hipHostMalloc(&roiTensorPtrSrc, noOfImages * sizeof(RpptROI));
+    hipHostMalloc(&roiTensorPtrDst, noOfImages * sizeof(RpptROI));
+
+    // Initialize the ImagePatch for source and destination
+    RpptImagePatch *srcImgSizes, *dstImgSizes;
+    hipHostMalloc(&srcImgSizes, noOfImages * sizeof(RpptImagePatch));
+    hipHostMalloc(&dstImgSizes, noOfImages * sizeof(RpptImagePatch));
+
+    // Set ROI tensors types for src/dst
+    RpptRoiType roiTypeSrc, roiTypeDst;
+    roiTypeSrc = RpptRoiType::XYWH;
+    roiTypeDst = RpptRoiType::XYWH;
+
+    // Set maxHeight, maxWidth and ROIs for src/dst
+    const int images = noOfImages;
+    string imageNames[images];
+
+    DIR *dr1 = opendir(src);
+    while ((de = readdir(dr1)) != NULL)
+    {
+        if (strcmp(de->d_name, ".") == 0 || strcmp(de->d_name, "..") == 0)
+            continue;
+        imageNames[count] = de->d_name;
+        string temp = "";
+        temp = src1;
+        temp += imageNames[count];
+        if (layoutType == 0 || layoutType == 1)
+            image = imread(temp, 1);
+
+        else
+            image = imread(temp, 0);
+
+        roiTensorPtrSrc[count].xywhROI.xy.x = 0;
+        roiTensorPtrSrc[count].xywhROI.xy.y = 0;
+        roiTensorPtrSrc[count].xywhROI.roiWidth = image.cols;
+        roiTensorPtrSrc[count].xywhROI.roiHeight = image.rows;
+
+        roiTensorPtrDst[count].xywhROI.xy.x = 0;
+        roiTensorPtrDst[count].xywhROI.xy.y = 0;
+        roiTensorPtrDst[count].xywhROI.roiWidth = image.cols;
+        roiTensorPtrDst[count].xywhROI.roiHeight = image.rows;
+
+        srcImgSizes[count].width = roiTensorPtrSrc[count].xywhROI.roiWidth;
+        srcImgSizes[count].height = roiTensorPtrSrc[count].xywhROI.roiHeight;
+        dstImgSizes[count].width = roiTensorPtrDst[count].xywhROI.roiWidth;
+        dstImgSizes[count].height = roiTensorPtrDst[count].xywhROI.roiHeight;
+
+        maxHeight = RPPMAX2(maxHeight, roiTensorPtrSrc[count].xywhROI.roiHeight);
+        maxWidth = RPPMAX2(maxWidth, roiTensorPtrSrc[count].xywhROI.roiWidth);
+        maxDstHeight = RPPMAX2(maxDstHeight, roiTensorPtrDst[count].xywhROI.roiHeight);
+        maxDstWidth = RPPMAX2(maxDstWidth, roiTensorPtrDst[count].xywhROI.roiWidth);
+
+        count++;
+    }
+    closedir(dr1);
+
+    // Check if any of maxWidth and maxHeight is less than or equal to 0
+    if(maxHeight <= 0 || maxWidth <= 0)
+    {
+        std::cerr<<"Unable to read images properly.Please check the input path of the files specified";
+        exit(0);
+    }
+
+    // Set numDims, offset, n/c/h/w values for src/dst
+    srcDescPtr->numDims = 4;
+    dstDescPtr->numDims = 4;
+
+    srcDescPtr->offsetInBytes = 64;
+    dstDescPtr->offsetInBytes = 0;
+
+    srcDescPtr->n = noOfImages;
+    srcDescPtr->h = maxHeight;
+    srcDescPtr->w = maxWidth;
+    srcDescPtr->c = inputChannel;
+
+    dstDescPtr->n = noOfImages;
+    dstDescPtr->h = maxDstHeight;
+    dstDescPtr->w = maxDstWidth;
```

Comment:
To consolidate the whole "// Set numDims, offset, n/c/h/w values for src/dst" section and other sections that are related to tensor descriptor initializations, can we have a helper called rpp_tensor_descriptor_initializer(RpptDescPtr, ...all params...)?
This function is called once for src and once for dst to finish initializing everything.

Reply:
We can discuss over this in a call

Reply:
Done